src/coreclr/vm/method.hpp

```diff
@@ -55,17 +55,27 @@ EXTERN_C VOID STDCALL NDirectImportThunk();
 #define METHOD_TOKEN_RANGE_BIT_COUNT (24 - METHOD_TOKEN_REMAINDER_BIT_COUNT)
 #define METHOD_TOKEN_RANGE_MASK ((1 << METHOD_TOKEN_RANGE_BIT_COUNT) - 1)
 
-enum class AsyncMethodType
+enum class AsyncMethodKind
 {
+    // Regular methods, including Task-returning ones.
+    // Those that do not return Task/ValueTask do not get a thunk sibling
     NotAsync,
-    TaskToAsync,
-    AsyncToTask,
-    Async
+
+    // Task-returning methods marked as async in IL.
+    // The method body forwards to Async2 implementation thunk
+    Async,
+
+    // Synthetic Async2 method that contains the
+    // actual implementation transformed into a resumable state machine
+    AsyncImplHelper,
+
+    // Synthetic Async2 method that forwards to the NotAsync Task-returning method
+    AsyncThunkHelper
 };
 
 struct AsyncMethodData
 {
-    AsyncMethodType type;
+    AsyncMethodKind type;
```

Comment:
Should this be renamed as well then?

Reply:
Yeah. I missed a few places. Mostly was looking at test failures after things started failing after the change.